.gitignore

```
@@ -1,9 +1,23 @@
+# General ignore patterns
 *
 !*.*
 !*/
+
+# Ignore build artifacts and temporary files
 *.o
 *.so
 *.a
 *.asv
 *~
-doc*/
+
+# Ignore Sphinx build output but include source files
+doc/build/
+doc/source/.*
+!doc/source/*.rst
+!doc/source/conf.py
+
+# Include all non-build documentation files
+!doc/
```

Hi @pritkc , thank you for your work. I would just remove these lines and for now leave only your line 14, where we exclude the `doc/build` path. 

You also want to exclude the path that you have specified in your Doxyfile:  in the `HTML_OUTPUT` variable, that is `HTML_OUTPUT            = doc/api_docs/cpp`

```suggestion
doc/api_docs/cpp
```